tests/unittest/executor/test_proxy_profile_sync.py

```
@@ -0,0 +1,191 @@
+        return ("start", None)  # ("status", optional_error_message)
+
+    proxy.profile_ack_queue.get.side_effect = blocking_get
+
+    def worker():
+        time.sleep(worker_setup_duration)
+        ack_event.set()
+
+    threading.Thread(target=worker, daemon=True).start()
+
+    t0 = time.monotonic()
+    proxy.start_profile(output_dir="/tmp/x", num_steps=5)
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/tmp/x"

(S108)